### DISCUSSION TITLE
My next lint doesn't show me errors that I encounter in next build or vercel

### DISCUSSION THREAD

**Discussion by asharrahman02** [vercel/next.js#63317] [Help] [answered]
### Summary

When working on my webstorm ide, I don't see errors in the editor, nor do I see errors when I run next lint. I have a eslint file that subscribes to core vitals and it seems to lint some things. However, it never catches the errors that fail my online vercel builds such as forgetting to omit the '.tsx' when importing components.

Any help would be appreciated, maybe I have something configured wrong? Or maybe I understand the types of errors that next lint will catch.

### Additional information

Here is what's in my ```eslintrc.json```

`
{
  "extends": "next/core-web-vitals"
}
`

**Comment by icyJoseph** [comment]
Can you run `npx next lint`? Locally? Does that give you the same errors? If so, then maybe your text editor is not correctly configured.

**Reply by asharrahman02** [reply]
Hi, Thanks for the reply @icyJoseph 

```npx next lint``` will output this:

```
./src/app/contributions/components/TransactionsList/TransactionsList.tsx
146:5  Warning: React Hook useEffect has missing dependencies: 'props.accountId' and 'props.userEmail'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
```

This is fine... but this is not an error that breaks my vercel build. This is the error I get when i build on vercel:

```
<html>
<body>
<!--StartFragment-->


Failed to compile.
05:18:35.867 |  
05:18:35.868 | ./src/app/api/stripe/account_session/route.ts:41:37
05:18:35.868 | Type error: 'error' is of type 'unknown'.
05:18:35.868 |  
05:18:35.868 | 39 \| 	} catch (error) {
05:18:35.868 | 40 \| 		console.error('An error occurred when calling the Stripe API to create an account session', error);
05:18:35.868 | > 41 \| 		return NextResponse.json({ error: error.message }, { status: 500 })
05:18:35.868 | \| 		                                  ^
05:18:35.868 | 42 \| 	}
05:18:35.868 | 43 \| }
05:18:35.938 | Error: Command "npm run build" exited with 1
05:18:36.435

<br class="Apple-interchange-newline"><!--EndFragment-->
</body>
</html>
```

my understanding is that ```npx next lint``` should catch these errors before i deploy to vercel so that  I can take care of them before hand. Furthermore, if my IDE/Lint is configured properly, I should see these errors in the text editor itself.

**Reply by icyJoseph** [reply, marked as answer]
Isn't that a Typescript error? The captured exception in try/catch blocks is now typed as unknown. It used to be any. 

So you have to make sure that your IDE is using your project's Typescript version. 

That'll highlight the error in your editor, but won't fix it. 

You can disable this Typescript behavior with: https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables set to false, but that's not good enough IMO.

You should verify that the caught exception is `instanceof Error` and then you can the read message out of it. 

To see this in your machine, run the build command in your package.json ~

**Reply by asharrahman02** [reply]
Thanks for your help @icyJoseph ! It seems it was because I used //@ts-ignore in one of my files for testing at some point and it seems to apply that globally. Not sure why but thank you regardless. I have removed it and it works fine now!
